internal/policies/mount/mount_test.go

```diff
@@ -0,0 +1,172 @@
+		computer   bool
+
+		usrDirPerm os.FileMode
+
+		secondCall      string
+		userReturnedUID string
+		userReturnedGID string
+
+		wantErr bool
+	}{
+		"successfully generates mounts file for one entry with one value": {},
+
+		"successfully generates mounts file for multiple entries with one value":       {entries: "multiple entries with one value"},
+		"successfully generates mounts file for one entry with multiple values":        {entries: "one entry with multiple values"},
+		"successfully generates mounts file for multiple entries with multiple values": {entries: "multiple entries with multiple values"},
+		"successfully generates mounts file for one entry with repeatead values":       {entries: "one entry with repeatead values"},
+		"successfully generates mounts file for multiple entries with the same value":  {entries: "multiple entries with the same value"},
+		"successfully generates mounts file for multiple entries with repeated values": {entries: "multiple entries with repeated values"},
+		"successfully generates mounts file for errored entries":                       {entries: "errored entries"},
+
+		"generates an empty file if the entry is empty": {entries: "one entry with no value"},
+		"generates no file if there are no entries":     {entries: "no entries"},
+
+		// Policy refresh.
+		"mount file is removed on refreshing policy with no entries": {secondCall: "no entries"},
+		"mount file is updated on refreshing policy with one entry":  {secondCall: "one entry with multiple values"},
+
+		// Error cases.
+		"error when user is not found":              {objectName: "dont exist", wantErr: true},
+		"error when user has invalid uid":           {userReturnedUID: "invalid", wantErr: true},
+		"error when user has invalid gid":           {userReturnedGID: "invalid", wantErr: true},
+		"error when usrDir has invalid permissions": {usrDirPerm: 0100, wantErr: true},
+
+		// To be removed when computer policies get implemented.
+		"error when trying to apply computer policies": {computer: true, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			testRunDir := t.TempDir()
+
+			opts := []mount.Option{mount.WithRunDir(testRunDir)}
+
+			usr, _ := user.Current()
+			if tc.objectName == "" {
+				if tc.userReturnedUID == "" {
+					tc.userReturnedUID = usr.Uid
+				}
+				if tc.userReturnedGID == "" {
+					tc.userReturnedGID = usr.Gid
+				}
+
+				tc.objectName = "ubuntu"
+
+				opts = append(opts, mount.WithUserLookup(func(string) (*user.User, error) {
+					return &user.User{Uid: tc.userReturnedUID, Gid: tc.userReturnedGID}, nil
+				}))
+			}
+
+			entries := mount.EntriesForTests["one entry with one value"]
+			if tc.entries != "" {
+				entries = mount.EntriesForTests[tc.entries]
+			}
+
+			if tc.usrDirPerm != 0 {
+				opts = append(opts, mount.WithPerm(0100))
+			}
+
+			m, err := mount.New(opts...)
+			require.NoError(t, err, "Expected no error but got one.")
```

Comment:
Same, this is a `Setup:` at this level as it’s not what we are testing exactly.

TODO: we should all come with a convention to ensure this is done everywhere, because it’s so easy to miss one…

Reply:
Indeed. I agree that this should be standardized.